src/StreamingDataFrames/streamingdataframes/platforms/quix/config.py

```diff
@@ -210,6 +239,64 @@ def get_workspace_ssl_cert(self, extract_to_folder: Optional[Path] = None) -> st
                 )
         return full_path.as_posix()
 
+    def _create_topic(self, topic: Topic):
+        cfg = topic.creation_configs
+        topic_name = self.strip_workspace_id(topic.name)
+        # an exception is raised (status code) if topic is not created successfully
+        self.api.post_topic(
+            topic_name=topic_name,
+            workspace_id=self.workspace_id,
+            topic_partitions=cfg.num_partitions,
+            topic_rep_factor=cfg.replication_factor,
+            topic_ret_bytes=cfg.optionals.get("retention.bytes"),
+            topic_ret_minutes=cfg.optionals.get("retention.ms", 0) // (60 * 1000)
+            or None,
+        )
+        logger.info(f"Creation of topic {topic_name} acknowledged by broker.")
+
+    def create_topics(self, topics: Iterable[Topic]):
+        logger.info("Attempting to create topics...")
+        current_topics = {t["id"] for t in self.get_topics()}
+        finalize = set()
+        for topic in topics:
+            if topic.name in current_topics:
+                logger.debug(
+                    f"topic {self.strip_workspace_id(topic.name)} already exists."
+                )
+            else:
+                self._create_topic(topic)
+                finalize.add(topic.name)
+        logger.info(
+            "Topic creations acknowledged; waiting for 'Ready' statuses..."
+            if finalize
+            else "No topic creations required!"
+        )
+        while finalize:
```

Comment:
Should we maybe add some timeout here? 
In case the topic creation takes more than expected.

Reply:
BTW, does the API class set any request timeouts at all?

Reply:
Added